code/datums/components/food/decomposition.dm

```diff
@@ -0,0 +1,162 @@
+//"Don't leave food on the floor, that's how we get ants"
+
+#define DECOMPOSITION_TIME 10 MINUTES
+#define DECOMPOSITION_TIME_RAW 5 MINUTES
+#define DECOMPOSITION_TIME_GROSS 7 MINUTES
+
+/datum/component/decomposition
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Makes sure food only starts decomposing if a player's EVER picked it up before
+	var/handled = FALSE
+	/// Used to stop food in someone's hand & in storage slots from decomposing.
+	var/protected = FALSE
+	/// Used to stop the timer & check for the examine proc
+	var/timerid
+	/// Used so the timer won't reset.
+	var/time_remaining = DECOMPOSITION_TIME
+	/// Used to give raw/gross food lower timers
+	var/decomp_flags
+	/// Used for examining
+	var/examine_type = 0
+
+/datum/component/decomposition/Initialize(decomp_flags = NONE)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.decomp_flags = decomp_flags
+
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/handle_movement)
+	RegisterSignal(parent, list(
+		COMSIG_ITEM_PICKUP, //person picks up an item
+		COMSIG_STORAGE_ENTERED), //Object enters a storage object (boxes, etc.)
+		.proc/picked_up)
+	RegisterSignal(parent, list(
+		COMSIG_ITEM_DROPPED, //Object is dropped anywhere
+		COMSIG_STORAGE_EXITED), //Object exits a storage object (boxes, etc)
+		.proc/dropped)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine) // Self-explanitory
+
+	if(decomp_flags & GROSS && !(decomp_flags & RAW))
+		time_remaining = DECOMPOSITION_TIME_GROSS
+		examine_type = 1
+	if(decomp_flags & RAW) // Raw food overrides gross
+		time_remaining = DECOMPOSITION_TIME_RAW
+		examine_type = 2
```

Comment:
Why not "if raw, do x, else if gross do y?

Reply:
that genuinely slipped my mind, my bad. Fixing!